### PR TITLE
Add support for Docker EE 2.0 support - Fix Issue #358

### PR DESCRIPTION
**Pull request by wdurairaj**
This PR introduces an addition parameter called `mount_prefix` which when set uses this directory as a base directory for the "Mountpoint" of a docker volume. 

Default directory of /opt/hpe/data/ is not visible in the ucp-kubelet container volume mapping and hence the flexvolumedriver fails to perform a mount for pod running in Docker EE 2.0 UCP environment.

### CODE REVIEW

File: hpedockerplugin/fileutil.py [hunk 1]
```diff
@@ -79,6 +79,13 @@ def mkdir_for_mounting(path):
             uuid = data[3]
         else:
             uuid = data[4]
+
+        if mount_prefix:
+            global prefix
```
Comment (imran-ansari) [inline hpedockerplugin/fileutil.py]: If mount_prefix is supported per backend, then using global variable can be troublesome when parallel mount requests with different backends having different mount_prefix come in. Why don't we just use mount_prefix instead which is safe?

File: hpedockerplugin/hpe/hpe3par_opts.py [hunk 2]
```diff
@@ -11,6 +11,9 @@
                default='',
                help="3PAR username with the 'edit' role",
                deprecated_name='hp3par_username'),
+    cfg.StrOpt('mount_prefix',
```
Comment (imran-ansari) [inline hpedockerplugin/hpe/hpe3par_opts.py]: Do we need to support this per backend or per host? If per host then we can remove it from here.

File: hpedockerplugin/volume_manager.py [hunk 3]
```diff
@@ -1432,7 +1432,12 @@ def _mount_volume(driver):
         # Determine if we need to mount the volume
         if vol_mount == volume.DEFAULT_MOUNT_VOLUME:
             # mkdir for mounting the filesystem
-            mount_dir = fileutil.mkdir_for_mounting(device_info['path'])
+            if self._hpepluginconfig.mount_prefix:
+                mount_prefix = self._hpepluginconfig.mount_prefix
```
Comment (imran-ansari) [inline hpedockerplugin/volume_manager.py]: if 'mount_prefix' is per host then 'host_config.mount_prefix' must be used.

File: hpedockerplugin/hpe/hpe3par_opts.py (hunk 2 above)
Comment (wdurairaj) [reply hpedockerplugin/hpe/hpe3par_opts.py]: It's per host (across backends)

File: hpedockerplugin/volume_manager.py (hunk 3 above)
Comment (wdurairaj) [reply hpedockerplugin/volume_manager.py]: Done. Fixed as per the review comment in the new commit.

**Comment by wdurairaj** [comment]
@imran-ansari  , you review now and merge if it's ok.

File: hpedockerplugin/volume_manager.py (hunk 3 above)
Comment (wdurairaj) [reply hpedockerplugin/volume_manager.py]: i will correct it. Thanks for catching that :-)

**Pull request merged**